lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -150,106 +133,215 @@ extension ASTGenVisitor {
 // Misc visits.
 // TODO: Some of these are called within a single file/method; we may want to move them to the respective files.
 extension ASTGenVisitor {
-  public func visit(_ node: MemberBlockItemSyntax) -> ASTNode {
-    visit(Syntax(node.decl))
+  public func generate(_ node: MemberBlockItemSyntax) -> ASTNode {
+    generate(node.decl)
   }
 
-  public func visit(_ node: InitializerClauseSyntax) -> ASTNode {
-    visit(node.value)
+  public func generate(_ node: InitializerClauseSyntax) -> ASTNode {
+    generate(node.value)
   }
 
-  public func visit(_ node: ConditionElementSyntax) -> ASTNode {
-    visit(node.condition)
+  public func generate(_ node: ConditionElementSyntax) -> ASTNode {
+    generate(Syntax(node.condition))
```

Comment:
this needed the `Syntax` upcast even with the `generate` functions below

Reply:
You shouldn’t need to if you add

```swift
  func generate(_ node: some SyntaxChildChoices) -> ASTNode {
    // This call recurses without disambiguation.
    return self.generate(Syntax(node))
  }
```